### PR TITLE
Integration/parallelization test scripts now place test scripts & artifacts into subdirectories of the root test folder

### PR DESCRIPTION
**Pull request by yantosca**
Based on comments from PR #1565, we have updated the GCC and GCHP integration/parallelization test scripts so that all components needed for integration tests are separated into multiple subdirectories in the root test folder.  The root test folder now looks like:
```console
$ ls
bin/  build/  CodeDir@  env/  logs/  README.md  rundirs/  scripts/
```
A README.md file is now placed in this folder in order to describe the contents of each subdirectory (the names of which are fairly self-explanatory).  The directory structure is identical for:

- GEOS-Chem Classic integration tests
- GEOS-Chem Classic parallelization tests
- GCHP integration tests

A new set of integration and parallelization tests are running.

### CODE REVIEW

**Comment by yantosca** [comment]
GEOS-Chem Classic integration tests all passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #44adbbe Merge branch 'docs/dev' into dev/14.1.0
GEOS-Chem #a0c9cf84b Further fixes in GCC integration & parallelization test scripts
HEMCO     #16541aa Update changelog for version release

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 41107082
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gc_05x0625_NA_47L_merra2_CH4........................Execute Simulation....PASS
gc_05x0625_NA_47L_merra2_fullchem...................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS40..................Execute Simulation....PASS
gc_4x5_merra2_aerosol...............................Execute Simulation....PASS
gc_4x5_merra2_carbon................................Execute Simulation....PASS
gc_4x5_merra2_CH4...................................Execute Simulation....PASS
gc_4x5_merra2_CO2...................................Execute Simulation....PASS
gc_4x5_merra2_fullchem..............................Execute Simulation....PASS
gc_4x5_merra2_fullchem_aciduptake...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_APM..........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_benchmark....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA_SVPOA.............Execute Simulation....PASS
gc_4x5_merra2_fullchem_LuoWd........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_marinePOA....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_RRTMG........................Execute Simulation....PASS
gc_4x5_merra2_Hg....................................Execute Simulation....PASS
gc_4x5_merra2_metals................................Execute Simulation....PASS
gc_4x5_merra2_POPs_BaP..............................Execute Simulation....PASS
gc_4x5_merra2_tagCH4................................Execute Simulation....PASS
gc_4x5_merra2_tagCO.................................Execute Simulation....PASS
gc_4x5_merra2_tagO3.................................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers......................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers_LuoWd................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 26
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
GEOS-Chem Classic parallelization tests mostly passed, except in a few specialty simulations (see #1637).
```console
==============================================================================
GEOS-Chem Classic: Parallelization Test Results

GCClassic #44adbbe Merge branch 'docs/dev' into dev/14.1.0
GEOS-Chem #a0c9cf84b Further fixes in GCC integration & parallelization test scripts
HEMCO     #16541aa Update changelog for version release

1st run uses 24 OpenMP threads
2nd run uses 13 OpenMP threads
Number of parallelization tests: 23

Submitted as SLURM job: 41107068
==============================================================================
 
Parallelization tests:
------------------------------------------------------------------------------
gc_05x0625_NA_47L_merra2_CH4........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....FAIL
gc_4x5_merra2_aerosol...............................Execute Simulation....PASS
gc_4x5_merra2_carbon................................Execute Simulation....PASS
gc_4x5_merra2_CH4...................................Execute Simulation....PASS
gc_4x5_merra2_fullchem..............................Execute Simulation....PASS
gc_4x5_merra2_fullchem_aciduptake...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_APM..........................Execute Simulation....FAIL
gc_4x5_merra2_fullchem_benchmark....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA_SVPOA.............Execute Simulation....PASS
gc_4x5_merra2_fullchem_LuoWd........................Execute Simulation....FAIL
gc_4x5_merra2_fullchem_marinePOA....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_RRTMG........................Execute Simulation....PASS
gc_4x5_merra2_Hg....................................Execute Simulation....FAIL
gc_4x5_merra2_metals................................Execute Simulation....PASS
gc_4x5_merra2_POPs_BaP..............................Execute Simulation....PASS
gc_4x5_merra2_tagCH4................................Execute Simulation....PASS
gc_4x5_merra2_tagCO.................................Execute Simulation....PASS
gc_4x5_merra2_tagO3.................................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers......................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers_LuoWd................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Parallelization tests passed: 19
Parallelization tests failed: 4
Parallelization tests not yet completed: 0
```

File: test/GCClassic/integration/README.template.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  - Contains individual GEOS-Chem Classic run directories.
+
+`scripts`
+
+  - Contains the scripts that are used for running the GEOS-Chem Classic integration tests.  These are copied from the `test/GCClassic/integration` folder of the GEOS-Chem "science codebase" repository.
+
+## For more information
+
+Please see the README.md file in the `test/GCClassic/integration`
```
Comment (lizziel) [inline test/GCClassic/integration/README.template.md:44]: Could the README that describes how to run the tests be copied into the scripts directory? If yes, this text can be updated to "Please see the README.md file in the 'scripts' subdirectory for detailed instructions on how to run GEOS-Chem Classic integration tests". Same comments for GCHP and parallelization READMEs.

**Review by lizziel — CHANGES_REQUESTED** [review]
Looks great! I just have one comment on the READMEs.

File: test/GCClassic/integration/README.template.md (hunk 1 above)
Comment (yantosca) [reply test/GCClassic/integration/README.template.md]: Thanks @lizziel.  Now updated in commit 2fd00748

**Review by lizziel — APPROVED** [review]
Thanks Bob! This looks great.

**Comment by yantosca** [comment]
GCHP integration tests finished as well:
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #47405e5 Update CMakeLists.txt for GCHP testing (geoschem/geos-chem PR #1565)
GEOS-Chem #1f4a3532f Add README.template.md in the test/GCHP/integration folder
HEMCO     #16541aa Update changelog for version release

Number of execution tests: 0

Submitted as SLURM job: 41108966
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_merra2_fullchem................................Execute Simulation....PASS
gchp_merra2_fullchem_benchmark......................Execute Simulation....PASS
gchp_merra2_fullchem_RRTMG..........................Execute Simulation....PASS
gchp_merra2_tagO3...................................Execute Simulation....FAIL
gchp_merra2_TransportTracers........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 4
Execution tests failed: 1
Execution tests not yet completed: -5
```

NOTE: The error in tests not completed is caused by this line in `integrationTestExecute.sh`:
```bash
numTests=$(count_rundirs "${itRoot}")
```
but this should be
```console
numTests=$(count_rundirs "${rundirsDir}")
```
I will push a fix for this.

**Pull request merged**